packages/core-data/src/selectors.ts

```diff
@@ -450,22 +454,36 @@ export const getRawEntityRecord = createSelector(
 			name,
 			key
 		);
-		return (
-			record &&
-			Object.keys( record ).reduce( ( accumulator, _key ) => {
-				if (
-					isRawAttribute( getEntityConfig( state, kind, name ), _key )
-				) {
-					// Because edits are the "raw" attribute values,
-					// we return those from record selectors to make rendering,
-					// comparisons, and joins with edits easier.
-					accumulator[ _key ] = record[ _key ]?.raw ?? record[ _key ];
-				} else {
-					accumulator[ _key ] = record[ _key ];
+		if ( ! record ) {
+			return;
+		}
+		const output = Object.keys( record ).reduce( ( accumulator, _key ) => {
+			if (
+				isRawAttribute( getEntityConfig( state, kind, name ), _key )
+			) {
+				// Because edits are the "raw" attribute values,
+				// we return those from record selectors to make rendering,
+				// comparisons, and joins with edits easier.
+				accumulator[ _key ] = record[ _key ]?.raw ?? record[ _key ];
+			} else {
+				accumulator[ _key ] = record[ _key ];
+			}
+			return accumulator;
+		}, {} as any );
+
+		if ( ! output.blocks ) {
+			if ( output.content && typeof output.content !== 'function' ) {
```

Comment:
I guess checking for a function is not really necessary in this context. Also probably not needed to check for the block property because it will never be there.